internal/database/users.go

```diff
@@ -1222,7 +1231,8 @@ SELECT u.id,
        u.tos_accepted,
        u.searchable,
        ARRAY(SELECT email FROM user_emails WHERE user_id = u.id AND verified_at IS NOT NULL) AS emails,
-       (SELECT account_id FROM user_external_accounts WHERE user_id=u.id AND service_type = 'scim') AS scim_external_id
+       (SELECT account_id FROM user_external_accounts WHERE user_id=u.id AND service_type = 'scim') AS scim_external_id,
+       (SELECT account_data FROM user_external_accounts WHERE user_id=u.id AND service_type = 'scim') AS scim_account_data
   FROM users u %s`
```

Comment:
I think you could use a `LEFT JOIN` instead and change the referenced part to `u.deleted_at IS NULL` instead of what it does now. That way, the query wil be much less complex IMO:
```suggestion
  FROM users u`
  LEFT JOIN user_external_accounts AS ua ON ua.user_id = u.id AND service_type = 'scim'
  %s
```

Reply:
That would be nice but it didn't work, unfortunately. See this comment: https://github.com/sourcegraph/sourcegraph/pull/48291#discussion_r1119175162

Reply:
Well, you said it would not be elegant, not that it did not work. I think it will be elegant actually 😅